### PR TITLE
add Javascript compatibility checks

### PR DESCRIPTION
**Pull request by rogpeppe**
This tests the node.js macaroon package for compatibility against
gopkg.in/macaroon.v1 and libmacaroons.
There may well be a more elegant way to do this :-)

### CODE REVIEW

File: compat_test.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func (*suite) TestSignature(c *gc.C) {
 		c.Logf("test %d: %s", i, test.about)
 		checkConsistency(c, func(pkg mcompat.Package) (interface{}, error) {
 			m := makeMacaroon(pkg, test.macaroon)
-			c.Logf("made macaroon %s with sig %x", macStr(m), m.Signature())
+			//c.Logf("made macaroon %s with sig %x", macStr(m), m.Signature())
```
Comment (mhilton) [inline compat_test.go]: I'm not keen on having commented out code. please remove it if it isn't needed or replace it.

File: js/interp.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+var sys = require("sys");
+var stdin = process.openStdin();
+var currentBuf = new Buffer(0);
+var state = {}
+stdin.on("data", function(d) {
+    var i, line, result;
+    currentBuf = Buffer.concat([currentBuf, d]);
+    for(i = 0; i < currentBuf.length; i++){
+        if(d[i] === 10){
```
Comment (mhilton) [inline js/interp.js:27]: Could you please add a comment saying 10 is '\n'?

File: compat_test.go (hunk 1 above)
Comment (frankban) [inline compat_test.go]: +1

File: js/interp.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+// A simple way to run javascript from Go.
+// The protocol is:
+//    - read a line from stdin
+//    - decode as base64
+//    - evaluate it
+//    - write back the result as a single line of base-64-encoded JSON containing
+//    an object {result, exception}.
+
+var sys = require("sys");
```
Comment (frankban) [inline js/interp.js:18]: New line to separate "imports" from code?

**Comment by mhilton** [comment]
:+1: LGTM with some comments.

**Comment by frankban** [comment]
:+1: with suggestions from Martin and the following questions:
- do we depend on nodejs for those tests? How dependencies are set up?
- don't we use errgo here?
- would you like to add license headers to all the files?
  Thanks for the branch Roger!

File: js/interp.js (hunk 2 above)
Comment (rogpeppe) [inline js/interp.js:27]: Done

File: js/interp.js (hunk 3 above)
Comment (rogpeppe) [inline js/interp.js:18]: Done

**Comment by rogpeppe** [comment]
Dependency setup - I think I'd prefer to leave that to another PR.

Errgo - I can't get excited about using errgo here. I don't think it's important
in small projects, particularly ones like this which aren't designed to be
used anywhere else. That could always be changed later.

License headers: done

**Pull request merged**